lib/flying_penguin/duffel/client.ex

```diff
@@ -55,6 +35,41 @@ defmodule FlyingPenguin.Duffel.Client do
     end
   end
 
+  def parse_response(response) do
+    IO.inspect(response[:id], label: "offer request id")
+    %DuffelResponse{}
+    |> Map.put(:cabin_class, response[:cabin_class])
+    |> Map.put(:duffel_offers, parse_offers(response[:offers]))
+  end
+
+  defp parse_offers(offers) do
+    Enum.map(offers, fn offer ->
+      %DuffelOffer{}
+      |> Map.put(:base_amount, offer[:base_amount])
+      |> Map.put(:carrier, offer[:owner][:name])
+      |> Map.put(:departing_at, get_first_departure_time(offer))
+      |> Map.put(:arriving_at, get_final_arrival_time(offer))
+      |> Map.put(:number_of_stops, get_stops(offer))
+    end)
+  end
+
+  defp get_stops(offer) do
+    [ slice | _] = offer.slices
+    length(slice.segments) - 1
+  end
+
+  defp get_first_departure_time(offer) do
+    [ slice | _] = offer.slices
+    [segment | _] = slice.segments
+    segment[:departing_at]
+  end
+
+  defp get_final_arrival_time(offer) do
+    [ slice | _] = offer.slices
+    [_ | segment] = slice.segments # this assumes that the segments are in chronological order. I'm not sure if that's the case
+    segment[:arriving_at]
+  end
+
   defp request_body(search_params) do
```

Comment:
Nitpick, but I think `build_request_body` (instead of `build_request`) would be more precise (line 73)

Reply:
agreed